idseq_pipeline/commands/host_filtering_functions.py

```diff
@@ -820,47 +842,34 @@ def truncate_input(fastq_file):
         t.join()
         assert t.completed and not t.exception
 
-    # Downloaded files here are unzipped but not yet truncated.
-    fastq_files = execute_command_with_output("ls %s/*.%s" % (FASTQ_DIR, unzipped_file_extension)).rstrip().split("\n")
-    # Identify input files and characteristics
+    fastq_files = execute_command_with_output("ls %s/*.%s" % (FASTQ_DIR, FILE_TYPE)).rstrip().split("\n")
     if len(fastq_files) not in [1, 2]:
         write_to_log("Number of input files was neither 1 nor 2. Aborting computation.")
         return # only support either 1 file or 2 (paired) files
 
-    if max(initial_sizes) > 30*1000*1000*1000:
-        # Truncate very large files to the same number of reads.
-        truncators = []
-        for f in fastq_files:
-            t = MyThread(target=truncate_input, args=[os.path.join(FASTQ_DIR, f)])
-            t.start()
-            truncators.append(t)
-        for t in truncators:
-            t.join()
-            assert t.completed and not t.exception
-        if truncated_inputs[0]:
-            execute_command("echo %d | aws s3 cp - %s/%s" % (MAX_INPUT_READS * len(fastq_files), SAMPLE_S3_OUTPUT_PATH, INPUT_TRUNCATED_FILE))
-
     initial_file_type_for_log = "fastq" if "fastq" in FILE_TYPE else "fasta"
     if len(fastq_files) == 2:
         initial_file_type_for_log += "_paired"
 
+    # Instantiate a stats instance
+    stats = StatsFile(STATS_OUT, RESULT_DIR, None, SAMPLE_S3_OUTPUT_PATH)
+
     # Download total_reads.json input, if present.  This is only provided with post-filtered inputs,
     # where we don't have the reads prior to host filtering.
+    # Record total number of input reads
     try:
         stats_in = StatsFile(STATS_IN, RESULT_DIR, SAMPLE_S3_INPUT_PATH, SAMPLE_S3_OUTPUT_PATH)
         stats_in.load_from_s3()
         total_reads = stats_in.get_total_reads()
         assert total_reads == int(total_reads)
         write_to_log("Post-filtered input with {total_reads} original total reads.".format(total_reads=total_reads))
     except:
+        total_reads = None
         stats_in = None
-        total_reads = count_reads(fastq_files[0], initial_file_type_for_log)
-        write_to_log("Unfiltered input with {total_reads} total reads.".format(total_reads=total_reads))
-
-    # Record total number of input reads
-    stats = StatsFile(STATS_OUT, RESULT_DIR, None, SAMPLE_S3_OUTPUT_PATH)
-    stats.data.append({'total_reads': total_reads})
+        write_to_log("Unfiltered input. Need host filtering")
 
+    if total_reads: # set total reads if available
```

Comment:
`if total_reads == None` would be better, because 0 is a valid total_reads but evaluates to False as an if guard.